D3D12HelloWorld/Graphics/CommandList.cs

```diff
@@ -148,14 +185,10 @@ public sealed class CommandList : IDisposable {
             mCommandList.IASetIndexBuffer(indexBufferView);
         }
 
-        public void SetPipelineState(PipelineState pipelineState) {
-            if (pipelineState.IsCompute) {
-                mCommandList.SetComputeRootSignature(pipelineState.RootSignature);
+        public void SetPipelineState(PipelineState pipelineState, bool setRootSignature) {
+            if (setRootSignature) {
+                SetRootSignature(pipelineState);
```

Comment:
Now that this functionality has been separated out, reuse it to avoid code duplication, and introduce a parameter to opt out of doing the two operations together.